samcli/local/docker/container.py

```diff
@@ -94,6 +96,24 @@ def create(self):
             "tty": False
         }
 
+        if self._delve_path:
+            LOG.info("in golang debug mode")
+            kwargs["volumes"] = {
```

Comment:
Design feedback: `container.py` is the base class for all containers. The options you add, like PTrace and delve mounts, are specific to Lambda Containers which is implemented in `lambda_containers.py` file. Can you try and move these changes to the `lambda_containers.py` file inside a method like `_additional_container_options` that is called from base-class here? In the subclass, you can add these properties only for golang in debug mode so it doesn't impact other runtimes.